libs/core/langchain_core/messages/v1.py

```diff
@@ -0,0 +1,566 @@
+        tool_call_chunks=tool_call_chunks,
+        response_metadata=response_metadata,
+        usage_metadata=usage_metadata,
+        id=chunk_id,
+    )
+
+
+@dataclass
+class HumanMessage:
+    """A message from a human user.
+
+    Represents input from a human user in a conversation, containing text
+    or other content types like images.
+
+    Attributes:
+        id: Unique identifier for the message.
+        content: List of content blocks containing the user's input.
+        name: Optional human-readable name for the message.
+        type: Message type identifier, always "human".
+    """
+
+    id: str
+    content: list[types.ContentBlock]
+    name: Optional[str] = None
+    """An optional name for the message.
+
+    This can be used to provide a human-readable name for the message.
+
+    Usage of this field is optional, and whether it's used or not is up to the
+    model implementation.
+    """
+    type: Literal["human"] = "human"
+    """The type of the message. Must be a string that is unique to the message type.
+
+    The purpose of this field is to allow for easy identification of the message type
+    when deserializing messages.
+    """
+
+    def __init__(
+        self, content: Union[str, list[types.ContentBlock]], id: Optional[str] = None
+    ):
+        """Initialize a human message.
+
+        Args:
+            content: Message content as string or list of content blocks.
+            id: Optional unique identifier for the message.
+        """
+        self.id = _ensure_id(id)
+        if isinstance(content, str):
+            self.content = [{"type": "text", "text": content}]
+        else:
+            self.content = content
+
+    def text(self) -> str:
+        """Extract all text content from the message.
+
+        Returns:
+            Concatenated string of all text blocks in the message.
+        """
+        return "".join(
+            block["text"] for block in self.content if block["type"] == "text"
+        )
+
+
+@dataclass
+class SystemMessage:
+    """A system message containing instructions or context.
+
+    Represents system-level instructions or context that guides the AI's
+    behavior and understanding of the conversation.
+
+    Attributes:
+        id: Unique identifier for the message.
+        content: List of content blocks containing system instructions.
+        type: Message type identifier, always "system".
+    """
+
+    id: str
+    content: list[types.ContentBlock]
+    type: Literal["system"] = "system"
+
+    def __init__(
+        self, content: Union[str, list[types.ContentBlock]], *, id: Optional[str] = None
+    ):
+        """Initialize a system message.
+
+        Args:
+            content: System instructions as string or list of content blocks.
+            id: Optional unique identifier for the message.
+        """
+        self.id = _ensure_id(id)
+        if isinstance(content, str):
+            self.content = [{"type": "text", "text": content}]
```

Comment:
for each case where we have `self.content = [{"type": "text", "text": content}]`, could we use the approach where we construct the TypedDict and then convert to dict? e.g.:

```suggestion
content_typed = TextContentBlock(type="text", text=content)
self.content = [content_typed.asdict()]

        if isinstance(content, str):
            self.content = [{"type": "text", "text": content}]
```